[CIAPP-1198] Remove APP_ANALYTICS from tests instrumentation

AdrianLC opened the pull request:
We don't use this on ci-app backend and it's actually harmful due to some internals it makes use lose some tests that go to APM instead

ericmustin commented:
👋 @AdrianLC , if i understand correctly, this would create a breaking change for folks+customers using these instrumentations, which we generally work hard to avoid unless necessary or stability related, or try to deprecate gracefully.  If you aren't using `analytics_enabled` for these integrations, is there a reason you cannot set `analytics_enabled: false`?

AdrianLC commented:
> 👋 @AdrianLC , if i understand correctly, this would create a breaking change for folks+customers using these instrumentations, which we generally work hard to avoid unless necessary or stability related, or try to deprecate gracefully. If you aren't using `analytics_enabled` for these integrations, is there a reason you cannot set `analytics_enabled: false`?

Hello Eric, I'll explain better. 

I work on CI-App backend where we have changed standard APM processing to treat these spans differently. One of the differences is we changed a service that handle spans not marked as app_analytics, but this doesn't work if the spans have this tag. So this is already broken at the moment for users, because spans don't end up where they should.

As far as I know these integrations were added specifically for the CI-App and I would not expect users to be using them outside of it. If they do, it will change for them when get out of beta anyway.

marcotc reviewed (approved):
I think it's ok to remove these, given these are new features, and this change actually makes it work correctly in the backend.

jirikuncar reviewed (commented):
👍

ericmustin commented:
> As far as I know these integrations were added specifically for the CI-App and I would not expect users to be using them outside of it. If they do, it will change for them when we get out of beta anyway.

Sounds good, fwiw in the future we can verify this using internal reporting tools.
